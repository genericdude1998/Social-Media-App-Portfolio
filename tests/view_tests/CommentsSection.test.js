import React from 'react';
import { shallow } from 'enzyme';
import ConnectedCommentSection from '../../src/view/Feed/Comments/CommentSection';
import {mockComments, mockId } from '../mockValues'

let wrapper;

jest.spyOn(React, 'useContext').mockImplementation(() => {return{
    theme: 'light',
}})

wrapper = shallow(
    <ConnectedCommentSection 
        comments={mockComments} 
        postId={mockId} 
        openComments={false} 
        onOpenComments={jest.fn()} 
        onCloseComments={jest.fn()}
    />
);

describe('CommentsSection', () => {
    it('should match snapshot when closed comments', () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('should match snapshot when open comments', () => {
        wrapper.setProps({openComments:true})
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('should match snapshot when light', () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('should match snapshot when dark', () => {
        wrapper.setProps({theme: 'dark'});
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('should match snapshot when given a 0 length comments array', () => {
        wrapper.setProps({comments:[]});
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
