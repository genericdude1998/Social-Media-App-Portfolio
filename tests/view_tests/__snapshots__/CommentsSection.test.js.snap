// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsSection should match snapshot when closed comments 1`] = `
<div
  className="baseCommentsSection null"
>
  <WithThemeHOC
    postId={0}
  />
  <ToggleComments
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    onCloseComments={[MockFunction]}
    onOpenComments={[MockFunction]}
    openComments={false}
    postId={0}
  />
</div>
`;

exports[`CommentsSection should match snapshot when dark 1`] = `
<div
  className="baseCommentsSection dark"
>
  <WithThemeHOC
    postId={0}
  />
  <ToggleComments
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    onCloseComments={[MockFunction]}
    onOpenComments={[MockFunction]}
    openComments={true}
    postId={0}
  />
  <WithThemeHOC
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    postId={0}
  />
</div>
`;

exports[`CommentsSection should match snapshot when given a 0 length comments array 1`] = `
<div
  className="baseCommentsSection dark"
>
  <WithThemeHOC
    postId={0}
  />
  <WithThemeHOC
    comments={Array []}
    postId={0}
  />
</div>
`;

exports[`CommentsSection should match snapshot when light 1`] = `
<div
  className="baseCommentsSection null"
>
  <WithThemeHOC
    postId={0}
  />
  <ToggleComments
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    onCloseComments={[MockFunction]}
    onOpenComments={[MockFunction]}
    openComments={true}
    postId={0}
  />
  <WithThemeHOC
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    postId={0}
  />
</div>
`;

exports[`CommentsSection should match snapshot when open comments 1`] = `
<div
  className="baseCommentsSection null"
>
  <WithThemeHOC
    postId={0}
  />
  <ToggleComments
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    onCloseComments={[MockFunction]}
    onOpenComments={[MockFunction]}
    openComments={true}
    postId={0}
  />
  <WithThemeHOC
    comments={
      Array [
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "1",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "2",
          "name": "mock_name",
        },
        Object {
          "content": "mock_content",
          "date": "mock_date",
          "id": "3",
          "name": "mock_name",
        },
      ]
    }
    postId={0}
  />
</div>
`;
